spring-integration-gcp/src/main/java/org/springframework/integration/gcp/outbound/PubSubMessageHandler.java

```diff
@@ -29,15 +37,28 @@
 
 	private final PubSubOperations pubSubTemplate;
 
+	private MessageConverter messageConverter;
+
 	private String topic;
 
 	public PubSubMessageHandler(PubSubOperations pubSubTemplate, String topic) {
 		this.pubSubTemplate = pubSubTemplate;
 		this.topic = topic;
+
+		StringMessageConverter stringMessageConverter = new StringMessageConverter();
+		stringMessageConverter.setSerializedPayloadClass(String.class);
```

Comment:
This property isn't use during conversion from `Message`

Reply:
Right, I'm using it here just so `toMessage()` adds `String` payload, not `byte[]`.

To make simple code like this possible:

```
  @Bean
  @ServiceActivator(inputChannel = "pubsubInputChannel")
  public MessageHandler messageReceiver() {
    return message -> {
      LOGGER.info("Message arrived here! Payload: " + message.getPayload());
    };
  }
```

Reply:
??? We talk here about `PubSubMessageHandler`, not `PubSubInboundChannelAdapter`